Nos/Service/NamesAPI.swift

```diff
@@ -120,53 +144,89 @@ class NamesAPI {
         throw Error.unexpected
     }
 
-    /// Makes a request to `/.well-known/nostr.json` at the host with the provided
-    /// username and matches the server result with the provided public key.
+    /// Makes a request to `/.well-known/nostr.json` at the host with the
+    /// provided username and matches the server result with the provided public
+    /// key.
     private func ping(
         username: String,
         host: URL,
         publicKey: PublicKey
     ) async throws -> PingResult {
         let request = URLRequest(
-            url: host.appending(queryItems: [URLQueryItem(name: "name", value: username)])
+            url: host.appending(
+                queryItems: [URLQueryItem(name: "name", value: username)]
+            )
         )
         let (data, response) = try await URLSession.shared.data(for: request)
         if let response = response as? HTTPURLResponse {
             let statusCode = response.statusCode
             if statusCode == 404 {
                 return .notFound
-            } else if statusCode == 200, let json = try JSONSerialization.jsonObject(with: data) as? [String: Any] {
-                let names = json["names"] as? [String: String]
-                let npub = names?[username]
-                if npub == publicKey.hex {
-                    return .match
-                } else {
-                    return .mismatch
+            } else if statusCode == 200 {
+                let jsonObject = try JSONSerialization.jsonObject(with: data)
+                if let json = jsonObject as? [String: Any] {
+                    let names = json["names"] as? [String: String]
+                    let npub = names?[username]
+                    if npub == publicKey.hex {
+                        return .match
+                    } else {
+                        return .mismatch
+                    }
                 }
             }
         }
         return .unableToPing
     }
 
-    private func buildURLRequest(url: URL, method: HTTPMethod, json: Data?, keyPair: KeyPair) throws -> URLRequest {
+    private func buildURLRequest(
```

Comment:
I did add these to NamesAPI, the code was inside another function though and we realized it would look nicer in its own function.

I don't know how we arrived to this, but by looking at the code, the function is very general and could be reused by other parts of the code. I wonder if the way we the tags
```
let tags = [
     ["u", url.absoluteString],
     ["method", method.rawValue]
]
```
we send are specific to how @dcadenas designed the endpoint, or it is something general to nostr. If it is the latter, we can refactor this so this code is more visible and can be re-used.